packages/react-dom/src/client/ReactDOMComponent.js

```diff
@@ -308,7 +308,10 @@ function setInitialDOMProperties(
         setInnerHTML(domElement, nextHtml);
       }
     } else if (propKey === CHILDREN) {
-      if (typeof nextProp === 'string') {
+      if (enableFloat && isHostSingletonType(tag)) {
+        // Host Singletons may contain non-react children so we cannot treat singleton Text
+        // children as properties
```

Comment:
While technically correct, I feel like these are not worth special casing.

What's the case where this happens? It makes no sense on `<head>` or `<html>`. You could have an empty box with just some text, but then why would you end up with other content like styles streaming in later?

There's not much third parties can do with that anyway.